File: Python/operations_with_sets.py
def process_2arrays(arr1, arr2):
	arr1_solo = set(arr1)
	arr2_solo = set(arr2)

	def exists_in_both():
		count = 0
		for a in arr1:
			if a in arr2_solo:
				count += 1
				arr1_solo.remove(a)
				arr2_solo.remove(a)
		return count

	bothCount = exists_in_both()

	def exists_in_one():
		return len(arr1) + len(arr2) - bothCount * 2

	return [bothCount, exists_in_one(), len(arr1_solo), len(arr2_solo)]

File: JavaScript/operations_with_sets.js
function process2Arrays(arr1, arr2) {
	let arr1Set = new Set(arr1);
	let arr2Set = new Set(arr2);

	// Set intersections contain only items from both sets
	let intersection = new Set([...arr1Set].filter(x => arr2Set.has(x)));

	// Set difference contain only items from first list, not in second list
	let diffArr1 = new Set([...arr1Set].filter(x => !arr2Set.has(x)));
	let diffArr2 = new Set([...arr2Set].filter(x => !arr1Set.has(x)));

	return [intersection.size, diffArr1.size + diffArr2.size, diffArr1.size, diffArr2.size];
}